h/views/api/groups.py

```diff
@@ -184,28 +174,19 @@ def remove_member(context: GroupContext, request):
     permission=Permission.Group.MEMBER_ADD,
     description="Add the user in the request params to a group.",
 )
-def add_member(context: GroupContext, request):
+def add_member(context: GroupMembershipContext, request):
     """
     Add a member to a given group.
 
     :raise HTTPNotFound: if the user is not found or if the use and group
       authorities don't match.
     """
-    user_svc = request.find_service(name="user")
     group_members_svc = request.find_service(name="group_members")
 
-    try:
-        user = user_svc.fetch(request.matchdict["userid"])
-    except InvalidUserId as err:
-        raise HTTPNotFound() from err
-
-    if user is None:
-        raise HTTPNotFound()
-
-    if user.authority != context.group.authority:
+    if context.user.authority != context.group.authority:
         raise HTTPNotFound()
 
-    group_members_svc.member_join(context.group, user.userid)
+    group_members_svc.member_join(context.group, context.user.userid)
```

Comment:
context contains the "target" of the request, add the target user to the target group.